### PR TITLE
feat(ctb): Start games at latest finalized output root

### PR DESCRIPTION
**Pull request by clabby**
## Overview

Alters the `FaultDisputeGame` to start from a more recent anchor than the genesis block.

### Changes

- The OptimismPortal holds the latest finalized output root in storage.
  - When the portal proxy is re-initialized during the upgrade, an anchor output root is set to kick things off for the first dispute game. This can be set to the latest finalized root in the L2OO when we do the upgrade, or we could keep it at genesis if we deploy early and get a game through.
- When a dispute game is initialized, it persists the latest output root from the portal into storage as its starting l2 output root. Once the game is initialized, it’s immutable just like it is now, so it won’t switch up on the challenger mid-flight.
  - When a dispute game resolves to DEFENDER_WINS (i.e. in favor of the root claim), it calls back to a new function in the OptimismPortal that is only callable by dispute games created by the DisputeGameFactory. This function checks that:
    - The caller is a game created by the factory,
    - The calling game was created after the respected game type was last updated
    - The calling game has resolved in favor of the root
    - The calling game’s l2 block number claim is greater than the current latest finalized root
    
    If all of the above passes, the portal updates the latest root, and future games use that as the starting anchor until a more up to date one comes along.

### CODE REVIEW

**Comment by clabby** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/9808?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#9808** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/9808?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#9810** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/9810?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `develop`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @clabby and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/contracts-bedrock/src/dispute/FaultDisputeGame.sol [hunk 1]
```diff
@@ -396,6 +394,12 @@ contract FaultDisputeGame is IFaultDisputeGame, Clone, ISemver {
         resolvedAt = Timestamp.wrap(uint64(block.timestamp));
 
         emit Resolved(status = status_);
+
+        // If the game resolved as `DEFENDER_WINS`, the root claim was deemed correct. Attempt to set the
+        // latest finalized output root.
+        if (status_ == GameStatus.DEFENDER_WINS) {
+            PORTAL.trySetLatestFinalizedOutputRoot();
```
Comment (clabby) [inline packages/contracts-bedrock/src/dispute/FaultDisputeGame.sol:401]: note: this doesn't respect the finalization delay for resolved games in the portal. This value is only used to set up the initial anchor in the games, and it's nice to have it set along with the global resolution, but we probably need to make this respect the delay and have an extra poke.

File: versions.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "abigen": "v1.10.25",
-  "foundry": "a170021b0e058925047a2c9697ba61f10fc0b2ce",
+  "foundry": "9f6bb3bb47de9d5a1f2a6c38cbc57e0f4f5508c2 ",
```
Comment (tynes) [inline versions.json:3]: Should this be its own PR?
